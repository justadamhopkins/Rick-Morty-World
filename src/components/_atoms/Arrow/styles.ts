import styled, { css } from "styled-components";

const left = `
 transform: rotate(135deg);
`;

const right = `
  transform: rotate(-45deg);
`;

const directions = {
  left,
};

export const StyledArrow = styled.i<{ direction: "left" | "right" }>`
  border: solid var(--white);
  border-width: 0 3px 3px 0;
  display: inline-block;
  padding: 3px;
  margin: 0 5px 2px 0px;
  ${({ direction }) =>
    directions[direction] &&
    css`
      ${directions[direction]}
    `};
`;
