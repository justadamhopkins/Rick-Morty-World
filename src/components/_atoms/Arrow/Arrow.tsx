import { FC } from "react";
import { StyledArrow } from "./styles";
interface Props {
  direction: "left" | "right";
}
export const Arrow: FC<Props> = ({ direction }) => {
  return <StyledArrow direction={direction}></StyledArrow>;
};

export default Arrow;
