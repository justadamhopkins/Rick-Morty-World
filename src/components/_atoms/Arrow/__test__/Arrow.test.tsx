import { render } from "@testing-library/react";
import { Arrow } from "../Arrow";

describe("Arrow", () => {
  const renderArrow = () => {
    return render(<Arrow direction="left" />);
  };
  test("should render an Arrow", () => {
    const { asFragment } = renderArrow();

    expect(asFragment()).toMatchSnapshot();
  });
});
