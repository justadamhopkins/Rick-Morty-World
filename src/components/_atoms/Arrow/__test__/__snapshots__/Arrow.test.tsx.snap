// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should render an Arrow 1`] = `
<DocumentFragment>
  <i
    class="styles__StyledArrow-sc-1st5r82-0 gfWSqF"
    direction="left"
  />
</DocumentFragment>
`;
